### PR TITLE
MLPAB-2237 - use secret in management account

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

Allow use of jwt key secret in the management account

Fixes MLPAB-2237

## Approach

- Add resource policy for secret
- add allowed roles variable to tfvars
- create a customer managed KMS key to allow for sharing across accounts
- add list of allow IAM roles/wildcards

## Learning

- https://docs.aws.amazon.com/secretsmanager/latest/userguide/auth-and-access_examples_cross.html

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+Fixes MLPAB-##
+
+## Approach
+
+Explain how your code addresses the purpose of the change
+
+## Learning
+
+Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Modernising LPA service
```
Comment (hawx) [inline docs/pull_request_template.md:13]: if we're going to have a template, should probably not reference "MLPAB-" and "Modernising LPA service"

File: terraform/account/data_sources.tf [hunk 2]
```diff
@@ -2,10 +2,18 @@ data "aws_default_tags" "default" {
   provider = aws.eu_west_1
 }
 
-data "aws_region" "eu_west_1" {
-  provider = aws.eu_west_1
-}
+# data "aws_region" "eu_west_1" {
+#   provider = aws.eu_west_1
+# }
```
Comment (hawx) [inline terraform/account/data_sources.tf:7]: still want this commented out?

File: terraform/modules/lambda/iam.tf [hunk 3]
```diff
@@ -1,5 +1,7 @@
 resource "aws_iam_role" "lambda" {
-  name               = "lambda-${var.lambda_name}-${var.environment_name}-${data.aws_region.current.name}"
+  name = "lambda-${var.lambda_name}-${var.environment_name}-${data.aws_region.current.name}"
+  # this path will be used to grant permission to the lambda to access the KMS key
+  # path               = "/lpa-store-lambda/"
```
Comment (hawx) [inline terraform/modules/lambda/iam.tf:4]: is this needed?

File: docs/pull_request_template.md (hunk 1 above)
Comment (andrewpearce-digital) [reply docs/pull_request_template.md]: thanks, updated

File: terraform/account/data_sources.tf (hunk 2 above)
Comment (andrewpearce-digital) [reply terraform/account/data_sources.tf]: removed

File: terraform/modules/lambda/iam.tf (hunk 3 above)
Comment (andrewpearce-digital) [reply terraform/modules/lambda/iam.tf:4]: it's useful for the PR that follows this one

**Review by hawx — APPROVED** [review]

**Pull request merged**